rero_ils/modules/stats/extensions.py

```diff
@@ -38,14 +38,19 @@ def pre_dump(self, record, data, dumper=None):
         :param data: the data to dump.
         :param dumper: the dumper class used to dump the record.
         """
+        # to filter the search list results
+        if record.get('config', {}).get('organisation'):
+            record['organisation'] = record['config']['organisation']
+
         if not current_librarian:
             return
 
         library_pids = current_librarian.library_pids
         if current_librarian.has_full_permissions:
             library_pids = current_librarian.organisation.get_libraries_pids()
 
-        record['values'] = list(filter(
-            lambda lib: lib['library']['pid'] in library_pids,
-            record['values']
-        ))
+        if record['type'] == 'librarian':
```

Comment:
I often prefer to use constant (stored into `model.py` file) instead of string... but it's very cosmetic
Maybe check `StatType` class ?